### PR TITLE
DBZ-626 Passing schema validator as AvroValidator instance instead of…

### PR DESCRIPTION
**Pull request by gunnarmorling**
… Function

https://issues.jboss.org/browse/DBZ-626

### CODE REVIEW

**Comment by gunnarmorling** [comment]
@jpechane rebased to latest master.

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresSchema.java [hunk 1]
```diff
@@ -48,7 +47,7 @@
     private final TableSchemaBuilder schemaBuilder;
     private final String schemaPrefix;
     private final Tables tables;
-    private final Function<String, String> schemaNameValidator;
+    private final AvroValidator schemaNameValidator;
```
Comment (jpechane) [inline debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresSchema.java]: Should not we remove the class and maybe? Validator is something that verifies that constraints are fullfilled for the value. But here we are not validating but actively converting the value to a valid one.

File: debezium-core/src/main/java/io/debezium/relational/TableSchemaBuilder.java [hunk 2]
```diff
@@ -55,9 +56,9 @@
      *
      * @param valueConverterProvider the provider for obtaining {@link ValueConverter}s and {@link SchemaBuilder}s; may not be
      *            null
-     * @param schemaNameValidator the validation function for schema names; may not be null
+     * @param schemaNameValidator the validator for schema names; may not be null
      */
-    public TableSchemaBuilder(ValueConverterProvider valueConverterProvider, Function<String, String> schemaNameValidator, Schema sourceInfoSchema) {
+    public TableSchemaBuilder(ValueConverterProvider valueConverterProvider, AvroValidator schemaNameValidator, Schema sourceInfoSchema) {
```
Comment (jpechane) [inline debezium-core/src/main/java/io/debezium/relational/TableSchemaBuilder.java]: Is it really correct? We are in this case replacing generic with specific? Maybe there should be a parent interface witch validate method that will be extended by AvroValidator. And the parameter should be of that parent interface type.

**Comment by gunnarmorling** [comment]
@jpechane I'm open for both your suggestions, in fact I just didn't do this as not to be derailed too much from the things I'm actually working on. Indeed the name isn't perfect, and I also like the idea of interface + Avro-based implementation.

How about `SchemaNameAdjuster` and `AvroSchemaNameAdjuster`?

**Comment by gunnarmorling** [comment]
@jpechane Rebased and added one more commit to do the discussed renaming.

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresSchema.java [hunk 3]
```diff
@@ -168,7 +167,7 @@ public Table tableFor(TableId id) {
     }
 
     protected String validateSchemaName(String name) {
```
[the remark this answers is not in the document]
Comment (gunnarmorling) [reply debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresSchema.java]: Ah, yes, good catch. Could you do the renaming yourself before merging? Thanks!

**Comment by jpechane** [comment]
@gunnarmorling Merged, thanks!

**Pull request merged**